jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -1614,7 +1614,12 @@ else if (!DispatcherType.REQUEST.equals(baseRequest.getDispatcherType()))
                         if (LOG.isDebugEnabled())
                             LOG.debug("Got Session ID {} from cookie {}", id, sessionCookie);
 
+                        //retrieve the session, which increments the reference count
                         HttpSession s = getHttpSession(id);
+                        //associate it with the request so its reference count
+                        //will be decremented as the request completes
+                        if (s != null && isValid(s))
+                            baseRequest.enterSession(s);
```

Comment:
In all other cases, the calls to `enterSession()` and `setSession()` are in 2 consecutive lines.
Is there a reason to split them in this case?

Reply:
I REALLY think we have to stop do "logic cleanups" in the middle of important time critical fixes.  
Such cleanups without clear reasons have broken things before.

Reply:
No, line 1622 cannot be moved down. The whole point of the PR is that when we receive a request that has multiple valid session id cookies, we need to associate both of them with the request so that the request will decrement the reference count on both of them on the way out (the ref count was incremented at line 1618 when each session is loaded).

If the requestedSessionId.equals(id) we don't assign the session object because both ids are the same, and thus the session object will be the same, thus no point in re-assigning it.